src/Push.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <vector>
+#include <string>
+#include <nanomsg/pipeline.h>
+#include "NanoMsg.h"
+#include "NanoProtocol.h"
+#include <memory>
+class Push {
+public:
+   Push(const std::string& location, const bool shouldConnect=true);
+   Push(const std::string&,
+        const int timeoutInMs,
+        const bool shouldConnect=true);
+   Push(const std::string&,
+        const int timeoutInMs,
+        const int bufferSize,
+        const bool shouldConnect=true);
+   std::string GetBinding() const;
+   void Send(const std::string& data, bool dontWait=false);
+   void Send(void * data, bool dontWait=false);
+   void Send(const std::vector<std::pair<void*, unsigned int>>& data,
+             bool dontWait=false);
+   virtual ~Push();
+   Push& operator=(const Push&) = delete;
+   Push(const Push&) = delete;
+   Push() = delete;
+private:
+   void SendMessage(NanoMsg& msg, bool dontWait=false);
```

Comment:
@KjellKod we should probably meet when i get back to discuss the API or even earlier if you guys are chomping at the bit to get it done, as now will be the best time to make any final API changes before fleshing out the pattern.